### PR TITLE
ENYO-2168: Marquee doesn't recalculate distance when scroller show/hide

### PR DESCRIPTION
**Pull request by KunmyonChoi**
scroll thumb.

Issue:
- The moon.Scroller doesn't have a way to detect content changes.
- So, we need to bubble up onRequestSetupBonds to update scroll boundary
  and thumb state.
- The problem is even if moon.Scroller (moon.ScrollStrategy) update it's
  scroll bounds and enable scroll thumb area. All the items inside
  doesn't get any notice from scroller when scroll thumb area is
  reserved by strategy (push padding-right).

Fix:
- It need to call resize on scroller when scroll thumb state is changed,
  because the scroll content area and item width is actually changed.

Signed-off-by: Kunmyon Choi <kunmyon.choi@lge.com>

### CODE REVIEW

File: lib/ScrollStrategy/ScrollStrategy.js [hunk 1]
```diff
@@ -807,6 +811,7 @@ var MoonScrollStrategy = module.exports = kind(
 		this.$.hColumn.addRemoveClass('v-scroll-enabled', enabled);
 		this.$.pageUpControl.spotlight = enabled && this.spotlightPagingControls;
 		this.$.pageDownControl.spotlight = enabled && this.spotlightPagingControls;
+		this.set('vEnabled', enabled);
```
Comment (aarontam) [inline lib/ScrollStrategy/ScrollStrategy.js]: Is there any reason we can't just call `resize` directly, instead of setting up observers?

Comment (Djspaceg) [inline lib/ScrollStrategy/ScrollStrategy.js]: I thought this same thing, but after a search discovered that `ScrollControls.js:updateScrollability` uses this property.
Does seem like a heavy approach...

Comment (ryanjduffy) [inline lib/ScrollStrategy/ScrollStrategy.js]: Notably, if both `vEnabled` and `hEnabled` change at the same time, `resize()` will be called twice. Might be worth a little extra code in `enableDisableScrollColumns` to ensure that doesn't happen.

**Comment by webOS101** [comment]
I went ahead and reimplemented this without the observers and the calls to .set().  While I really liked the observer approach at first I did decide it hides the connection between cause and effect and it is not as performant as a more direct approach.  I suppose this is one of those situations that flux deals with well.

**Pull request merged**